core/modules/replicate.cc

```diff
@@ -64,26 +64,17 @@ CommandResponse Replicate::CommandSetGates(
   return CommandSuccess();
 }
 
-void Replicate::ProcessBatch(bess::PacketBatch *batch) {
-  bess::PacketBatch out_gates[ngates_];
-  for (int i = 0; i < ngates_; i++) {
-    out_gates[i].clear();
-  }
-
+void Replicate::ProcessBatch(const Task *task, bess::PacketBatch *batch) {
   for (int i = 0; i < batch->cnt(); i++) {
     bess::Packet *tocopy = batch->pkts()[i];
-    out_gates[0].add(tocopy);
+    EmitPacket(task, tocopy, 0);
```

Comment:
We could stand to have a promise here that EmitPacket() doesn't drop the packet before we can copy it. I think that's _mostly_ true, but if we put a copy into a dead batch and the dead batch fills up, we'll call `deadend()`, which might free it prematurely here. The obvious alternative is to copy before calling `EmitPacket` here (which might be a better way to code this anyway).

Reply:
Great point!! Thanks!